.semaphore/semaphore.yml

```diff
@@ -0,0 +1,175 @@
+version: v1.0
+name: CI Pipeline
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu2004
+blocks:
+  - name: Test Bootstrap
+    dependencies: []
+    run:
+      when: change_in('/bootstrap/')
+    task:
+      jobs:
+        - commands:
+            - checkout
+            - cd bootstrap
+            - go test -v -coverprofile=coverage.out ./...
+            - go tool cover -html=coverage.out -o coverage.html 
+          name: Test Bootstrap
+  - name: Test certs
+    dependencies: []
+    run:
+      when: change_in('/certs/')
+    task:
+      jobs:
+        - commands:
+            - checkout
+            - cd certs
+            - go test -v -coverprofile=coverage.out ./...
+            - go tool cover -html=coverage.out -o coverage.html 
+          name: Test certs
+  - name: Test http
+    dependencies: []
+    run:
+      when: change_in('/http/api/')
+    task:
+      jobs:
+        - commands:
+            - checkout
+            - cd http/api
+            - go test -v -coverprofile=coverage.out
+            - go tool cover -html=coverage.out -o coverage.html 
+          name: Test http
+  - name: Test logger
+    dependencies: []
+    run:
+      when: change_in('/logger/')
+    task:
+      jobs:
+        - commands:
+            - checkout
+            - cd logger
+            - go test -v -coverprofile=coverage.out ./...
+            - go tool cover -html=coverage.out -o coverage.html 
+          name: Logger test
+  - name: Test all

```

Comment:
This doesn't seem to be correct. I'd suggest to test everything probably during merge to main or after a reviewer has commented `LGTM`

Reply:
There is probably no need for this test at all, even in cases we merge to `master`. However, we need to be sure we tun tests when it is relevant (e.g. when proto files change, we need to run almost all the tests).